syntax = "proto3";

import "google/protobuf/empty.proto";

option go_package = "proto/interest";

message CreateInterest {
        string name = 1;
        uint64 userId = 2;
}

message Interest {
        uint64 id = 1;
        string name = 2;
        uint64 userId = 3;
}

message InterestResponse {
        repeated Interest interests = 1;
}

message GetInterestsRequest {
        uint64 userId = 1; 
}

message AddInterestsRequest {
        repeated CreateInterest interests = 1;
        uint64 userId = 2;
}

message DeleteInterestsRequest {
        repeated uint64 ids = 1;
}

service InterestService {
        rpc AddInterests(AddInterestsRequest) returns (InterestResponse);
        rpc DeleteInterests(DeleteInterestsRequest) returns (google.protobuf.Empty);
        rpc GetInterests(GetInterestsRequest) returns (InterestResponse);
}

